src/utils/map-state.js

```diff
@@ -235,11 +231,7 @@ export default class MapState {
     // Make sure we zoom around the current mouse position rather than map center
     const startZoomLngLat = this._interactiveState.startZoomLngLat ||
       this._unproject(startPos) || this._unproject(pos);
-    let {startZoom} = this._interactiveState;
-
-    if (!Number.isFinite(startZoom)) {
-      startZoom = this._viewportProps.zoom;
-    }
+    const startZoom = ensureFinite(this._interactiveState.startZoom, this._viewportProps.zoom);
```

Comment:
Q: Why do we need to `ensureFinite here`? Aren't the `viewportProps` already vetted?  No need to answer, just a thought.

Reply:
`ensureFinite` checks `this._interactiveState.startZoom`, and falls back to `this._viewportProps.zoom` if it's undefined. This is because we have two modes of zoom - scroll zoom that are discrete events, and pinch zoom that are continuous events.

Reply:
> This is because we have two modes of zoom - scroll zoom that are discrete events (transform from the current zoom level), and pinch zoom that are continuous events (transform from the zoom level when pinch started).

Oh I see, then this is not a parameter validation. To explain what tripped me up, the use of `ensureFinite` here, while convenient, mislead me since normally I see that function being used for parameter validation, but here it is used for logic. I would have considered coding this differently, if not, I think your explanation above would have added value to the code in the form of a comment.

Reply:
+1 to adding this as a comment